app/components/application_component.rb

```diff
@@ -6,19 +6,20 @@ class ApplicationComponent < ViewComponent::Base
   include SvgHelper
   include ColorHelper
 
-  def initialize(styles: [], style: nil, **attrs)
```

Comment:
I was surprised that the `Card` component didn't have any padding. I noticed that we only use it in 3 places at the moment and it always has a nested `box` component whose only job is to add padding and/or background colors.

I do not like the idea of adding html elements for the purpose of styling as this can be done with CSS and scoping. I would like the ability to use the Card component, or any other as the parent element in my component and scope it to that particular component. Giving frontenders the ability to pass in a unique class name to scope the CSS is a nice addition to our component library.

I originally updated the Card component. See [`a9b23ec` (#1378)](https://github.com/tactilenews/100eyes/pull/1378/commits/a9b23ecb1f63eb0a1001b8fb5a45b8c84c8e268e) where I removed the code as it changed several components which have nothing to do with this feature.

I don't think there is any use case where we would want to use a card without padding and being able to configure its background color sound like something that would be super useful. I plan to put in another PR to reintroduce these changes, but I still feel like having the ability to pass down a class from one component to another is super useful. One of the things that drives me crazy about popular component libraries is their inflexibility in giving the ability to do so when you want to apply CSS to change some styling.

Reply:
> I see your point with the nested box component. On the other hand, using a box component for padding and/or background colours also is very versatile and removes the need to add padding and background parameters to every component. I assume that was/is the idea behind the box component.

I come from the school that says that we HTML elements are for structure and CSS to apply styling. We add HTML elements in this case only for the purpose of adding some styling, ie padding and background color when the same can be achieved with CSS. In the case that we want to dry something out, we can use @mixins to create a style and apply it widely. One way this affects us is that we have a lot of nested elements, some of which could be avoided by following this principle, and it makes it more difficult to navigate the HTML tree. Considering this example:

![Screenshot from 2022-09-21 10-50-44](https://user-images.githubusercontent.com/26943915/191459666-b055fef0-f7b3-4c3f-a5e1-fa57906dd2c3.png)

There are 11 nested elements! Maybe others don't mind this. I know that @tillprochaska has reasons for setting it up like this. I don't think one is wrong and the other right, I just think they are different ways of looking at frontend code and how it's organized. I know that he sent me a link explaining how he looks at it, but I can't seem to find it. Anyways, I think there is space for doing things in slightly different ways.

Thankfully, @tillprochaska already provided a way to inject classes and other HTML attributes. A PR I helped review and had forgotten about. See  [Add helpers to add HTML attributes to component #452 
](https://github.com/tactilenews/100eyes/pull/452). This means that we don't need to make any changes, so no follow-up PR for this.